source/jormungandr/jormungandr/scenarios/helper_classes/streetnetwork_path.py

```diff
@@ -80,27 +93,115 @@ def __init__(
         self._request_id = request_id
         self._async_request()
 
-    @new_relic.distributedEvent("direct_path", "street_network")
-    def _direct_path_with_fp(self):
-        with timed_logger(self._logger, 'direct_path_calling_external_service', self._request_id):
-            try:
-                return self._streetnetwork_service.direct_path_with_fp(
+    @staticmethod
+    def poi_to_pt_object(poi):
+        return type_pb2.PtObject(poi=poi, uri=poi.uri, embedded_type=type_pb2.POI, name=poi.name)
+
+    def get_pt_objects(self, entry_point):
+        if include_poi_access_points(self._request, entry_point, self._mode):
+            for o in entry_point.poi.children:
+                yield self.poi_to_pt_object(o)
+        else:
+            yield entry_point
+
+    def make_poi_access_points(self, fallback_type, best_dp_element):
+        entry_point = (
+            self._orig_obj if fallback_type == StreetNetworkPathType.BEGINNING_FALLBACK else self._dest_obj
+        )
+        if not include_poi_access_points(self._request, entry_point, self._mode):
```

Comment:
maybe add this check outside this function, and do not call this function when `include_poi_access_points` is false

Reply:
nevermind, I get it, you need the entry_point